File: src/main/java/com/simon/MindCrew/entity/SkillPack.java
package com.simon.MindCrew.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 技能包 · 提问时可选的"场景化能力"
 *   = 名称/图标 + 技能指令(角色/风格/输出) + 可选绑定知识库范围
 */
@Data
@TableName("skill_pack")
public class SkillPack {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    /** 图标(emoji 或图标名) */
    private String icon;

    private String description;

    /** 技能指令：告诉 AI 这个技能干啥、怎么答、什么风格/格式 */
    private String instruction;

    /** 绑定的知识库 collection id 列表(JSON 数组)；空=全部知识库 */
    private String collectionIds;

    /** 启用：1=是 0=否 */
    private Integer enabled;

    private Integer sortOrder;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;
}

File: MindCrew-frontend/src/api/skillPack.ts
import request from '@/utils/request'

export interface SkillPack {
  id?: number
  name: string
  icon?: string
  description?: string
  instruction?: string
  collectionIds?: string   // JSON 数组字符串，如 "[1,2]"
  enabled?: number
  sortOrder?: number
}

export const skillPackApi = {
  /** 销售端：可用技能包 */
  usable: (): Promise<any> => request.get('/v2/skill-pack/usable'),
  /** 管理员：全部 */
  list: (): Promise<any> => request.get('/v2/skill-pack/list'),
  create: (data: SkillPack): Promise<any> => request.post('/v2/skill-pack', data),
  update: (id: number, data: SkillPack): Promise<any> => request.put(`/v2/skill-pack/${id}`, data),
  setEnabled: (id: number, enabled: boolean): Promise<any> =>
    request.put(`/v2/skill-pack/${id}/enabled`, null, { params: { enabled } }),
  delete: (id: number): Promise<any> => request.delete(`/v2/skill-pack/${id}`),
}
